### DISCUSSION TITLE
Is it possible to use a WPF or UWP component library in a WinUI3 application?

### DISCUSSION THREAD

**Discussion by americoperez49** [microsoft/WindowsAppSDK#750] [Q&A] [answered]
Just wondering if this is possible or if components will have to be rebuilt from the ground up.

**Comment by ghost** [comment, marked as answer]
@americoperez49  

see the response from the PM of WinUI himself here : [What is the role of XAML Islands in WinUI 3?](https://github.com/microsoft/ProjectReunion/discussions/465#discussioncomment-395712)

> Note: Today you can't mix in the same app WinUI 2/UWP XAML and WinUI 3 XAML. And this valid for Islands as well. You can't have a WPF app using WinUI 2 (using system XAML Islands) and WinUI 3 (using WinUI 3 XAML Islands) in the same process.

**Reply by andrewleader** [reply]
Thanks @ecovio1 for posting that answer! That's indeed the correct answer. We hope to enable this with XAML Islands in the future, but it's not supported today.
